File: PBKit/PBCookieStorage.m
//
//  PBCookieStorage.m
//
//  Created by pebble8888 on 2015/05/02.
//  Copyright (c) 2015 pebble. All rights reserved.
//

#import "PBCookieStorage.h"

@implementation PBCookieStorage

- (void)setCookie:(NSHTTPCookie *)aCookie
{
    [[NSHTTPCookieStorage sharedHTTPCookieStorage] setCookie:aCookie];
}

- (void)setCookies:(NSArray *)aCookies
{
    for( NSHTTPCookie* cookie in aCookies ){
        [[NSHTTPCookieStorage sharedHTTPCookieStorage] setCookie:cookie];
    }
}

- (NSArray *)cookiesForURL:(NSURL *)aURL
{
    return [[NSHTTPCookieStorage sharedHTTPCookieStorage] cookiesForURL:aURL];
}

@end

File: PBKit/PBCookieStorage.swift
//
//  PBCookieStorage.swift
//
//  Created by pebble8888 on 2017/01/08.
//  Copyright © 2017年 pebble8888. All rights reserved.
//

import Foundation

protocol PBCookieStorageProtocol
{
    func cookiesForURL(url:URL) -> [HTTPCookie]
    func setCookies(cookies:[HTTPCookie])
}

class PBCookieStorage : NSObject, PBCookieStorageProtocol
{
    public func setCookie(cookie:HTTPCookie)
    {
        HTTPCookieStorage.shared.setCookie(cookie)
    }

    public func setCookies(cookies:[HTTPCookie])
    {
        for cookie in cookies {
            HTTPCookieStorage.shared.setCookie(cookie)
        }
    }

    public func cookiesForURL(url:URL) -> [HTTPCookie]
    {
        guard let cookies = HTTPCookieStorage.shared.cookies(for: url) else {
            return []
        }
        return cookies
    }
}
